### PR TITLE
Implement file storage

### PR DESCRIPTION
**Pull request by tekulvw**
The mechanism behind this is probably going to need to change but I'm not sure what it should be at the moment.

### CODE REVIEW

**Comment by tekulvw** [comment]
DO NOT MERGE UNTIL #77 HAS BEEN MERGED

**Review by JAKeeney — APPROVED** [review]
ok_emoji

**Pull request merged**
